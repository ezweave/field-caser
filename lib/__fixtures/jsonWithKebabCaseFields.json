{
	"first-name": "Joe",
	"last-name": "Consumer",
	"details": {
		"date-of-birth": "1/1/1970",
		"cars": [{
				"brand": "Pontiac",
				"model": "Aztek",
				"year-of-manufacture": "2004"
			},{
				"brand": "Saturn",
				"model": "Vue",
				"year-of-manufacture": "2004"
		}]
	}
}
