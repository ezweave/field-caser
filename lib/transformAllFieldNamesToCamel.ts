import { camelCase } from 'lodash'
import { deepFieldTransformer } from './deepFieldTransformer'

export const transformAllFieldNamesToCamel: <T>(source: unknown) => T = deepFieldTransformer(camelCase)
