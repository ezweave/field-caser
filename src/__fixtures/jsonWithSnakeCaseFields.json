{
	"first_name": "Joe",
	"last_name": "Consumer",
	"details": {
		"date_of_birth": "1/1/1970",
		"cars": [{
				"brand": "Pontiac",
				"model": "Aztek",
				"year_of_manufacture": "2004"
			},{
				"brand": "Saturn",
				"model": "Vue",
				"year_of_manufacture": "2004"
		}]
	}
}
