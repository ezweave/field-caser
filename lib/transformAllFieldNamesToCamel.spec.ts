import { transformAllFieldNamesToCamel } from './transformAllFieldNamesToCamel'
import * as jsonWithSnakeCaseFields from './__fixtures/jsonWithSnakeCaseFields.json'
import * as jsonWithKebabCaseFields from './__fixtures/jsonWithKebabCaseFields.json'

describe('transformAllFieldNamesToCamel', () => {
  it('transform snake field names to camelcase', () => {
    expect(transformAllFieldNamesToCamel(jsonWithSnakeCaseFields)).toMatchSnapshot()
  })
  it('transform kebab field names to camelcase', () => {
    expect(transformAllFieldNamesToCamel(jsonWithKebabCaseFields)).toMatchSnapshot()
  })
})
