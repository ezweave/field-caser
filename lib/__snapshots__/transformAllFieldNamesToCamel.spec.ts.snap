// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformAllFieldNamesToCamel transform kebab field names to camelcase 1`] = `
Object {
  "default": Object {
    "details": Object {
      "cars": Object {
        "0": Object {
          "brand": "Pontiac",
          "model": "Aztek",
          "yearOfManufacture": "2004",
        },
        "1": Object {
          "brand": "Saturn",
          "model": "Vue",
          "yearOfManufacture": "2004",
        },
      },
      "dateOfBirth": "1/1/1970",
    },
    "firstName": "Joe",
    "lastName": "Consumer",
  },
  "details": Object {
    "cars": Object {
      "0": Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "yearOfManufacture": "2004",
      },
      "1": Object {
        "brand": "Saturn",
        "model": "Vue",
        "yearOfManufacture": "2004",
      },
    },
    "dateOfBirth": "1/1/1970",
  },
  "firstName": "Joe",
  "lastName": "Consumer",
}
`;

exports[`transformAllFieldNamesToCamel transform snake field names to camelcase 1`] = `
Object {
  "default": Object {
    "details": Object {
      "cars": Object {
        "0": Object {
          "brand": "Pontiac",
          "model": "Aztek",
          "yearOfManufacture": "2004",
        },
        "1": Object {
          "brand": "Saturn",
          "model": "Vue",
          "yearOfManufacture": "2004",
        },
      },
      "dateOfBirth": "1/1/1970",
    },
    "firstName": "Joe",
    "lastName": "Consumer",
  },
  "details": Object {
    "cars": Object {
      "0": Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "yearOfManufacture": "2004",
      },
      "1": Object {
        "brand": "Saturn",
        "model": "Vue",
        "yearOfManufacture": "2004",
      },
    },
    "dateOfBirth": "1/1/1970",
  },
  "firstName": "Joe",
  "lastName": "Consumer",
}
`;
